File: src/main/java/inferno/client/resources/sounds/Sound.java
package inferno.client.resources.sounds;

import java.io.*;

public class Sound {
    private File audioFile;
    private String name;

    public Sound(String name) {
        try {
            this.audioFile = new File(this.getClass().getClassLoader().getResource("assets/test_game/sounds/" +name+".wav").getFile());
        } catch (NullPointerException e) {
            e.printStackTrace();
        }
        this.name = name;
    }

    public void play(){ }

    public String getName() {
        return name;
    }
}

File: src/main/kotlin/common/sounds/Sound.kt
package common.sounds

import common.registries.RegistryNameable
import java.io.File

class Sound(val name: String, val domain: String): RegistryNameable(name) {
    private var audioFile: File? = null

    init {

        this.javaClass.classLoader.getResource("assets/$domain/sounds/$name.wav")?.let {
            this.audioFile = File(it.file)
        } // TODO: Logger warn here if null.
    }

    fun play() {}
}
